### PR TITLE
Feature/26 crud dashboard

### PR DESCRIPTION
**Pull request by eddiejaoude**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,9 +3,13 @@ language: php
 php:
   - 5.5
 
+before_install:
+  - mysql -e "create database IF NOT EXISTS dashboardhub;" -uroot
```
Comment (brettminnie) [inline .travis.yml]: would it not be safer to not set this up as root and create a db user and assign?

File: src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    /**
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * @var string
+     */
+    protected $repo;
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php]: repo? funny you let that past seeing as you are so pedantic :)

Is repository not more accurate? Also will this ever be a collection?

Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php]: True, it should be `repository` :+1:

Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php]: And yes, good point, it could be a `Collection`

File: .travis.yml (hunk 1 above)
Comment (eddiejaoude) [inline .travis.yml]: Its just for CI, then it gets destroyed. Not sure if really worth it?

File: src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php (hunk 2 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php]: For now i might keep it simple. I will have a think & bounce ideas off you.

Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php]: Renamed it to `Repository` though.

Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Entity/Dashboard.php]: :+1:

File: app/config/parameters_dev.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # This file is auto-generated during the composer install
 parameters:
     database_driver: pdo_mysql
-    database_host: 192.168.59.103
+    database_host: 127.0.0.1
```
Comment (brettminnie) [inline app/config/parameters_dev.yml]: Does localhost not work for this?

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php [hunk 4]
```diff
@@ -10,8 +14,82 @@
...
+    /**
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
     public function indexAction()
     {
-        return $this->render('DashboardHubAppBundle:Dashboard:index.html.twig');
+        return $this->render(
+                'DashboardHubAppBundle:Dashboard:index.html.twig',
+                array(
+                    'dashboards' => $this->get('dashboardhub_app_main.service.dashboard')
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: formatting: if you split a call, all bits must be split

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php [hunk 5]
```diff
@@ -10,8 +14,82 @@
...
+                        ->findAllByAuthenticatedUserAndDefaults()
+                )
+            );
+    }
+
+    public function addAction(Request $request)
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: docblock

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php [hunk 6]
```diff
@@ -10,8 +14,82 @@
...
+        }
+
+        return $this->render('DashboardHubAppBundle:Dashboard:add.html.twig',
+                array(
+                    'form' => $form->createView()
+                )
+        );
+    }
+
+    public function editAction(Request $request, $id)
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: docblock

File: src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        $resolver->setDefaults(
+            array(
+                'data_class' => 'DashboardHub\Bundle\AppBundle\Entity\Dashboard'
+            )
+        );
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php]: docblock

File: src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            )
+            ->add(
+                'save',
+                'submit',
+                array('label' => 'Save')
+            );
+    }
+
+    public function getName()
+    {
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php]: docblock

File: app/config/parameters_dev.yml (hunk 3 above)
Comment (eddiejaoude) [inline app/config/parameters_dev.yml]: Yep, I changed it back to `localhost`

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php (hunk 4 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: Split a call?

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php [hunk 9]
```diff
@@ -10,8 +14,82 @@
...
+        $dashboard = new Dashboard();
+        $form = $this->createForm(new DashboardType(), $dashboard);
+
+        $form->handleRequest($request);
+
+        if ($form->isValid()) {
+            $dashboard = $form->getData();
+
+            $this->get('dashboardhub_app_main.service.dashboard')
+                ->save($dashboard);
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: formatting

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php [hunk 10]
```diff
@@ -10,8 +14,82 @@
...
+
+            $request->getSession()->getFlashBag()->add(
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: formatting

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php [hunk 11]
```diff
@@ -10,8 +14,82 @@
...
+                'success',
+                'Dashboard created'
+            );
+            return $this->redirect($this->generateUrl('dashboardhub_app_dashboard.list'));
+        }
+
+        return $this->render('DashboardHubAppBundle:Dashboard:add.html.twig',
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: formatting

File: src/DashboardHub/Bundle/AppBundle/Menu/MenuBuilder.php [hunk 12]
```diff
@@ -42,13 +42,8 @@ public function createMainMenu(Request $request)
...
         $menu->addChild('Home', array('route' => 'dashboardhub_app_homepage'));
         if ($this->securityContext->isGranted('IS_AUTHENTICATED_FULLY')) {
-            $menu->addChild(
-                $this->securityContext->getToken()->getUser()->getUsername(),
-                array('route' => 'dashboardhub_app_homepage')
-            );
             $menu->addChild('Dashboard', array('route' => 'dashboardhub_app_dashboard.list'));
-            $menu->addChild('Create', array('route' => 'dashboardhub_app_dashboard.create'));
-            $menu->addChild('Logout', array('route' => 'logout'));
+            $menu->addChild('Logout ' . $this->securityContext->getToken()->getUser()->getUsername(), array('route' => 'logout'));
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Menu/MenuBuilder.php]: formatting - long line

File: src/DashboardHub/Bundle/AppBundle/Service/DashboardService.php [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+/**
+ * Class Dashboard
+ * @package DashboardHub\Bundle\AppBundle\Service
+ */
+class DashboardService
+{
+    /**
+     * @var EntityManager
+     */
+    private $em;
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Service/DashboardService.php]: class not final, should this not be protected?

File: src/DashboardHub/Bundle/AppBundle/Service/DashboardService.php [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+
+    /**
+     * @var SecurityContext
+     */
+    private $securityContext;
+
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Service/DashboardService.php]: As above

File: src/DashboardHub/Bundle/AppBundle/Service/VersionService.php [hunk 15]
```diff
@@ -10,6 +11,7 @@ class VersionService
      * @var string
      */
     private $version;
+
```
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Service/VersionService.php]: as above

**Comment by brettminnie** [comment]
Some pendantic comments,

re the formatting

```php
$this
    ->method()
    ->method(
       arg1,
       arg2
    )
    ->method();
```

Is what I was referring to, you have a mix of split chained calls and non split ones

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php (hunk 4 above)
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: see main comment :)

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php (hunk 5 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: Fixed in next PR https://github.com/DashboardHub/PipelineDashboard/pull/64/files#diff-cbf79bbb80822cc4f7d69acaff41f11aR37

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php (hunk 6 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: Fixed in next PR https://github.com/DashboardHub/PipelineDashboard/pull/64/files#diff-cbf79bbb80822cc4f7d69acaff41f11aR70

File: src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php (hunk 7 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php]: Fixed in next PR https://github.com/DashboardHub/PipelineDashboard/pull/64/files#diff-31f6da7cd6ebb596c3498d3c5d9606ed

File: src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php (hunk 8 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Form/DashboardType.php]: Fixed in next PR https://github.com/DashboardHub/PipelineDashboard/pull/64/files#diff-31f6da7cd6ebb596c3498d3c5d9606ed

File: src/DashboardHub/Bundle/AppBundle/Service/DashboardService.php (hunk 13 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Service/DashboardService.php]: Good spot! :+1:

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php (hunk 10 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: I split into multiple lines to be consistent :smile: (in next PR though as the file has changed)

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php (hunk 11 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: I split this `$request->getSession()->getFlashBag()->add(` onto multiple lines, are you referring to anything else?

File: src/DashboardHub/Bundle/AppBundle/Menu/MenuBuilder.php (hunk 12 above)
Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Menu/MenuBuilder.php]: Yep too long. Sorted in next PR.

**Comment by eddiejaoude** [comment]
All updates were done in the next PR as I **refactored** a lot of the code https://github.com/DashboardHub/PipelineDashboard/pull/64

File: src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php (hunk 10 above)
Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: Does the next pr require this?

On Sun, Mar 8, 2015 at 10:26 PM, Eddie Abou-Jaoude <notifications@github.com

Comment (eddiejaoude) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: The next PR contains a refactor, so made sense to do it in that (e.g. form to services). Flash message still required.

Comment (brettminnie) [inline src/DashboardHub/Bundle/AppBundle/Controller/DashboardController.php]: Ok, I'll merge this then first

**Comment by brettminnie** [comment]
merged

**Pull request merged**